### ISSUE TITLE
Using `breakafterinrun` to break long sequences of dashes in tutorial

### ISSUE THREAD

**Issue by muzimuzhi** [T-F-S/tcolorbox#294]
In `tcolorbox-tutorial-poster.pdf`, page 13 (the last page)
- current
  ![image](https://github.com/user-attachments/assets/8436f84a-438a-43b6-ad6c-3d3029282996)
- proposed
  ![image](https://github.com/user-attachments/assets/2b0f8859-c4c6-4cfa-9f24-2dea1bd518f6)
  ```diff
  diff --git a/doc/latex/tcolorbox/tcolorbox-tutorial-poster.tex b/doc/latex/tcolorbox/tcolorbox-tutorial-poster.tex
  index ebc9b4e..13ea843 100644
  --- a/doc/latex/tcolorbox/tcolorbox-tutorial-poster.tex
  +++ b/doc/latex/tcolorbox/tcolorbox-tutorial-poster.tex
  @@ -1373,7 +1373,7 @@ Our poster is finished now. Just go to the next page to see the final result.
       \tcbinputlisting{standard jigsaw,size=minimal,toprule=0.5mm,bottomrule=0.5mm,boxsep=2mm,
         colback=white,opacityback=0.75,listing only,
         enforce breakable,tcb@poster@boxheight,before skip=-\interlineskip,height fixed for=all,
  -      minted options={tabsize=2,fontsize=\small,breaklines,breakafter={\,\]-}},
  +      minted options={tabsize=2,fontsize=\small,breaklines,breakafter={\,\]-},breakafterinrun},
         listing file=tcolorbox-example-poster.tex}%
     }
   \end{tcbposter}
  ```

Related: https://github.com/gpoore/fvextra/issues/27

**Comment by muzimuzhi** [comment]
BTW `minted` v3 doesn't need shell escape anymore.
https://github.com/T-F-S/tcolorbox/blob/2911a3ec729951f0d664ee761c76ac9cd16f3909/doc/latex/tcolorbox/tcolorbox-tutorial-poster.tex#L21-L22

`minted` v3 ships with a new script/executable `latexminted(.py)` which is auto trusted by TeX Live since [TeX Live r72375](https://tug.org/svn/texlive?view=revision&revision=72375). This can be verified in stdout of `tlmgr conf texmf shell_escape_commands` or `kpsewhich -var-value shell_escape_commands`.

**Comment by T-F-S** [comment]
Thank you. I will `breakafterinrun`.

BTW I had major problems with `minted` v3 during package building and finally switched to `minted2`
- `latexmk` never stopped, because `*.index.minted` changed constantly. I tried to construct a MWE, but could not find a reliable one to communicate (works sometimes, sometimes not).
- Umlauts did not work with `minted` v3 (had this problem with the `csvsimple` package)

**Comment by muzimuzhi** [comment]
> * `latexmk` never stopped, because `*.index.minted` changed constantly.

Is it a simple `latexmk tcolorbox.tex` or with some customized config? I thought that tcolorbox docs were compiled via `arara`.

**Comment by T-F-S** [comment]
> Is it a simple `latexmk tcolorbox.tex` or with some customized config? I thought that tcolorbox docs were compiled via `arara`.

I used a manual script and later `arara` to build the documentation, but it became too complex to figure out the needed number of compiler runs. So, I switched to `latexmk` some time ago to savely automate the needed reruns.

I made a MWE which does not need `tcolorbox` to fail with `minted`:

```latex
% !TeX encoding=UTF-8
\documentclass{article}
%\usepackage{minted2}
\usepackage{minted}
%\usepackage[minted]{tcolorbox}

\begin{document}

Test

\begin{minted}{tex}
Hello \textbf{world}
\end{minted}

%\begin{tcblisting}{}
%Hello \textbf{world}
%\end{tcblisting}

\end{document}
```

My `.latexmkrc` file is

```
$pdf_mode = 4;        # tex -> pdf mit lualatex
$lualatex = 'pdflatex -synctex=1 -interaction=batchmode -shell-escape';
$max_repeat = 16;
```

This is nearly identical to my tcolorbox setup where the increase of `max_repeat` is needed.

The problem with the MWE is that it does not fail every time. Minutes ago, I tried from vanilla several times (no fail), removed `*.fdb_latexmk`, compiled a single `pdflatex`, etc. to finally get the fail:

![grafik](https://github.com/user-attachments/assets/91ed02fb-6920-4986-92d9-3867b8d32d41)

This is a endless loop which stops with error after 16 reruns.

The more complex `tcolorbox` documentation always fails, but a (successfull) creation with `latexmk` takes about 30 minutes.

**Comment by muzimuzhi** [comment]
> Minutes ago, I tried from vanilla several times (no fail), removed `*.fdb_latexmk`, compiled a single `pdflatex`, etc. to finally get the fail:

On my side, after a manual single `pdflatex`, then `latexmk` always resulted in an endless loop.

The culprit is the timestamp stored in `*.index.minted` which is accurate to the second. The `.latexmkrc` below worked for me:
```perl
$pdf_mode = 4;        # tex -> pdf mit lualatex
$lualatex = 'pdflatex -synctex=1 -interaction=batchmode -shell-escape';
$max_repeat = 4;

$hash_calc_ignore_pattern{'minted'} = 'timestamp';
```
It seems `hash_calc_ignore_pattern` cannot handle double extensions, as both of `hash_calc_ignore_pattern('index.minted')` and `hash_calc_ignore_pattern('index\.minted')` didn't work.

`*.index.minted` is a new `minted` aux file (created by the corresponding `latexminted` python package and containing a json object) [since v3](https://github.com/gpoore/minted/blob/main/latex/CHANGELOG_MINTED_LATEX_PACKAGE.md#v300-20240922), thus there's little info about it across other docs.

I think this deserves to be reported to `minted`.

**Comment by T-F-S** [comment]
> ```perl
> $hash_calc_ignore_pattern{'minted'} = '"timestamp":';
> ```

Thank you - this seems to be exactly what I was looking for. I cannot test before the weekend, but this looks good!

**Comment by gpoore** [comment]
I think that the issue with the `*.index.minted` is that it is being updated more than necessary. I'm working on finalizing the next `minted` release, and I will change things so that the index file is no longer replaced unless the data it contains has changed.

**Comment by muzimuzhi** [comment]
> * `latexmk` never stopped, because `*.index.minted` changed constantly. I tried to construct a MWE, but could not find a reliable one to communicate (works sometimes, sometimes not).

Should be fixed in `latexminted` python package, [v0.2.0](https://github.com/gpoore/minted/releases/tag/python%2Fv0.2.0).

> * Umlauts did not work with `minted` v3 (had this problem with the `csvsimple` package)

Probably similar to https://github.com/gpoore/minted/issues/411 thus also fixed in `latexminted` [v0.2.0](https://github.com/gpoore/minted/releases/tag/python%2Fv0.2.0).

**Comment by T-F-S** [comment]
Done with https://github.com/T-F-S/tcolorbox/releases/tag/v6.4.1
